src/utils/reporting_workflow.js

```diff
@@ -0,0 +1,154 @@
+    currently being processed
+
+  cancelToken: indicates the job should be cancelled
+    (from a process external to the job itself -- this tells the job to cease processing)
+
+  exitStatus: if set, terminates the job and marks it with the given status
+    (for example, if a given patient doesn't meet reporting criteria, don't keep reporting)
+}
+
+Try to keep the wiki page up to date with this implementation!
+https://github.com/mcode/medmorph-backend/wiki/Reporting-Workflow
+*/
+
+const BASE_REPORTING_BUNDLE_PROFILE =
+  'http://hl7.org/fhir/us/medmorph/StructureDefinition/us-ph-reporting-bundle';
+
+/**
+ * IG_REGISTRY maps the profile for a reporting bundle under an IG
+ * to the actions defined for action codes within that IG.
+ * Each content IG should have one JS file defined, which, when require()d,
+ * will register itself into this IG.
+ * Note that the base actions should be extend-able, so that content IGs
+ * do not have to re-implement codes unnecessarily.
+ */
+const IG_REGISTRY = {
+  [BASE_REPORTING_BUNDLE_PROFILE]: {} // base action definitions, see MEDMORPH-35
+};
+
+function initializeContext(planDefinition) {
+  // TODO -- see MEDMORPH-36
+  return { id: uuidv4(), planDefinition };
+}
+
+function getFunction(ig, code) {
+  // use a function here in case this needs to be more complex
+  return IG_REGISTRY[ig][code];
+}
+
+/**
+ * Find the target profile to be used when generating a bundle
+ * in support of the given PlanDefinition.
+ * The profile should be on the 'output' field of the 'create-report' action.
+ */
+function findProfile(planDefinition) {
+  // figure out the profile of the target bundle
+
+  // TODO search arrays rather than hardcode 0,0
+  const createReportAction = planDefinition.action?.find(
+    a => a.code[0].coding[0].code === 'create-report'
+  );
+
+  if (!createReportAction) {
+    throw new Error(`PlanDefinition ${planDefinition.id} has no action with code 'create-report'`);
+  }
+
+  const bundleOutput = createReportAction.output?.find(o => o.type === 'Bundle');
+
+  if (!bundleOutput?.profile?.[0]) {
+    throw new Error(
+      `PlanDefinition ${planDefinition.id} does not specify a profile for report bundle`
+    );
+    // option B is just use the default profile
+  }
+
+  return bundleOutput.profile[0];
+}
+
+function determineActionSequence(planDefinition) {
+  // TODO:
+  // note: 6.1.8
+  // The Backend Service App SHALL implement the ability
+  // to process action dependencies specified via
+  // the PlanDefinition.action.relatedAction elements.
+  // Typical implementations will build a graph or a tree structure
+  // that can be used to identify dependencies between actions.
+
+  // simple version to just get the integers 0 -> n
+  // https://stackoverflow.com/a/10050831
+  return [...Array(planDefinition.action.length).keys()];
+}
+
+function executeWorkflow(context) {
+  if (!context.currentActionSequenceStep) {
+    context.currentActionSequenceStep = 0;
+  }
```

Comment:
Is there a reason this is done here and not in `initializeContext`?

Reply:
Good call - will move that